cmd/vochaintest/vochaintest.go

```diff
@@ -94,23 +95,44 @@ func main() {
 	log.Init(*loglevel, "stdout")
 	rand.Seed(time.Now().UnixNano())
 
-	// create entity key
-	entityKey := ethereum.NewSignKeys()
-	if len(*entityPrivKey) > 0 {
-		if err := entityKey.AddHexKey(*entityPrivKey); err != nil {
+	// create entity keys
+	entityKeys := make([]*ethereum.SignKeys, len(*entityPrivKeys))
+	for i, key := range *entityPrivKeys {
+		entityKeys[i] = ethereum.NewSignKeys()
+		if err := entityKeys[i].AddHexKey(key); err != nil {
+			log.Fatalf("error adding provided key %s", key)
+		}
+	}
+	// create oracle key
+	oracleKey := ethereum.NewSignKeys()
+	if len(*oraclePrivKey) > 0 {
+		if err := oracleKey.AddHexKey(*oraclePrivKey); err != nil {
 			log.Fatal(err)
 		}
 	} else {
-		if err := entityKey.Generate(); err != nil {
+		if err := oracleKey.Generate(); err != nil {
 			log.Fatal(err)
 		}
 	}
+	// create treasurer key
+	if len(*treasurerPrivKey) == 0 {
```

Comment:
We should not require always the treasurerPrivKey, just if its needed

Reply:
As some accounts must be always initialized, the treasurer must be able to send those transactions, without the treasurer the accounts start just empty and no test can be run.  I think we should always require the treasurer addr

Reply:
Only if it is required. Consider the caller knows what is doing. He might been funded the accounts previously.